src/main/java/org/savantbuild/dep/workflow/Workflow.java

```diff
@@ -153,7 +153,8 @@ public Path fetchSource(Artifact artifact) throws ProcessFailureException, MD5Ex
         // Publish the Savant named source JAR to prevent going back out to remote repositories next time we want to load source JARs
         if (file != null) {
           item = new ResolvableItem(artifact.id.group, artifact.id.project, artifact.id.name, artifact.version.toString(), artifact.getArtifactSourceFile());
-          publishWorkflow.publish(item, file);
+          // we should be returning our locally cached/published source path, not the remote source path
+          return publishWorkflow.publish(item, file);
```

Comment:
This method was designed to fall through all the way down. If you want to make this change, you can go ahead and clean up the rest of the method that is using the null state of file as well (i.e. line 161 doesn't need to check for null if I am reading this correctly).

Reply:
I thought I wrote back but don't see it here. I changed `return` to set the local `file` variable instead and thus be consistent with the existing method flow.